enterprise/views.py

```diff
@@ -1067,18 +1066,16 @@ def get(self, request):
         if tpa_hint:
             # Add the tpa_hint to the redirect's 'next' query parameter
             # Redirect will be to the Enterprise Customer's TPA provider
-            tpa_next_param = {
-                'tpa_hint': tpa_hint,
-            }
-            query_dict['next'] = update_query_parameters(str(query_dict['next']), tpa_next_param)
+            query_dict['tpa_hint'] = tpa_hint
```

Comment:
This looks like you are removing a level of indirection that was in use before. Is it possible to do a quick git blame and figure out why it was done this way initially? Maybe we need the value both encoded for the next URL level AND at this current level? (Unless the original coder was just mistaken on where the value went)

Reply:
this is a good question. I now wonder if the tpa_hint is _also_ needed in the 'next' url we are forwarding to... will check

Reply:
since @chavesj made a prior change, and looks like this will undo that, I am going to pull him and consult before making this change

Reply:
esp since this PR had a fix in which required the tpa_hint in the `next` url it looks like https://github.com/edx/edx-enterprise/pull/961/files

Reply:
After taking a look at [this commit](https://github.com/edx/edx-enterprise/commit/2f34adac7b2ad87b9e3a2d749be05e2aadce5270) it seems like the `tpa_hint` param is needed as part of the `next` query param. Based on the commit message it seems needed for the correct welcome template to be rendered after authenticating with TPA.

Reply:
Thanks @chavesj going to add that back as well so I don't break that use case. To be clear, I will also add it to the `next` url, but also maintain it when redirecting to the /login url as I am doing here.. .we on the same page?

Reply:
so plan is to add it to next as a urlencoded value like before, but also have tpa_hint be passed to the redirect_uri...

Reply:
so yeah it will look like this now 

```
http://lms.example.com/login?next=http%3A%2F%2Flocalhost%3A18000%3Ftpa_hint%3Didp-1&tpa_hint=idp-1
```

Reply:
please check latest commit